glotaran/model/megacomplex.py

```diff
@@ -1,12 +1,26 @@
 from __future__ import annotations
 
+from typing import TYPE_CHECKING
+
 import numpy as np
 
 from glotaran.model import DatasetDescriptor
 from glotaran.model import model_attribute
 
+if TYPE_CHECKING:
+    from typing import Any
+
+
+def megacomplex(
```

Comment:
Please add some documentation for the new megacomplex decorator or the MegaComplex class for that matter; it is an important concept that is at the heart of the current refactoring.

Reply:
Docstring added